contrib/pg_tde/documentation/docs/replication.md

```diff
@@ -1,4 +1,257 @@
+    * The standby, upon startup, uses these settings to load the key.
+
+### Scenario 2: HashiCorp Vault Key Provider**
+
+1. **Primary:**
+    * Ensure the principal key (e.g., `my_vault_principal_key`) exists in Vault at the configured path (for example, `secret/data/postgres/keys`).
+    * In `postgresql.conf`:
+
+        ```ini
+        pg_tde_key_provider = 'vault'
+        pg_tde_vault_url = 'http://your-vault-server:8200'
+        pg_tde_vault_token = 'your-vault-access-token' # Or use other auth methods
+        pg_tde_vault_mount_path = 'secret/' # Path where KV engine is mounted
+        # pg_tde_vault_ca_path = '/path/to/vault_ca.crt' # If using self-signed certs
+        pg_tde_principal_key_name = 'my_vault_principal_key' # Key name within Vault path
+        ```
+
+    * After restarting PostgreSQL, in `psql`:
+
+        ```sql
+        CREATE EXTENSION IF NOT EXISTS pg_tde;
+        -- Add the key definition (pg_tde will fetch it from Vault)
+        SELECT pg_tde_principal_key_add('my_vault_principal_key', NULL, 'vault');
+        -- Set this key as the active principal key
+        SELECT pg_tde_set_principal_key('my_vault_principal_key', 'vault');
+        ```
+
+2. **Standby:**
+    * In the standby's `postgresql.conf`, configure `pg_tde_key_provider` and all `pg_tde_vault_*` parameters, and `pg_tde_principal_key_name` to be **identical** to the primary's configuration.
+    * Ensure the standby server has network access to the Vault server and the necessary credentials/permissions to read the *same principal key* from Vault.
+    * The standby, upon startup, uses these settings to load the key.
+
+**Is it possible to use the same key provider and principal key?**
+
+**Yes, it is mandatory.** For physical replication to function with TDE, the standby *must* use the same principal key (by name and content, accessed via the same provider configuration) as the primary to decrypt the replicated data blocks.
+
+## Encrypted Table Replication
+
+**If a table on the primary site is encrypted, will all the standby sites also have the encrypted table replicated, given that the same key is implemented on the standby sites as well?**
+
+Yes.
+
+Physical replication works at the block level.
+
+1. When you encrypt a table on the primary using `CREATE TABLE ... USING heap_tde;` or by setting `default_table_access_method = 'heap_tde'`, the data blocks on disk are encrypted.
+2. Changes to this table (also encrypted) are written to WAL.
+3. Streaming replication sends these WAL records to the standby.
+4. The standby applies these WAL records, writing the encrypted blocks to its disk.
+5. If the standby has the same principal key available (via identical `pg_tde` configuration and key access), it can decrypt these blocks when you query the table on the standby.
+
+If the standby lacks the correct key, replicated data remains encrypted and unreadable, and queries will fail.
+
+## Setup Steps
```

Comment:
We had setup instructions with two different providers above, why is another setup section here now?